pymodbus/transport/transport.py

```diff
@@ -310,12 +311,26 @@ def data_received(self, data: bytes):
         self.datagram_received(data, None)
 
     def datagram_received(self, data: bytes, addr: tuple):
-        """Receive datagram (UDP connections)."""
+        """Receive datagram"""
```

Comment:
Why change this ?

Reply:
Per the commit message, it was confusing and incorrect. This path is used for serial clients.

Reply:
No that path is used for datagrams which are udp, but the data method which handles all other types calls this, in order not to duplicate code.

Please revert this change.